src/test/java/io/confluent/kafka/connect/datagen/DatagenConnectorTest.java

```diff
@@ -110,38 +111,66 @@ public void shouldAllowSettingSchemaFile() {
   @Test
   public void shouldFailValidationWithMultipleSchemaSources() {
     clearSchemaSources();
-    config.put(DatagenConnectorConfig.SCHEMA_STRING_CONF, "a schema");
-    config.put(DatagenConnectorConfig.SCHEMA_FILENAME_CONF, "a schema file");
-
+    config.put(DatagenConnectorConfig.SCHEMA_STRING_CONF,
+            "{\"namespace\":\"ksql\",\"name\":\"test_schema\",\"type\":\"record\",\"fields\":" +
+                    "[{\"name\":\"id\",\"type\":{\"type\":\"long\",\"arg.properties\"" +
+                    ":{\"iteration\":{\"start\":0}}}}]}");
+    config.put(DatagenConnectorConfig.SCHEMA_FILENAME_CONF, "product.avro");
     Config validated = connector.validate(config);
 
     assertThat(
         validated,
-        hasValidationError(
-            DatagenConnectorConfig.SCHEMA_STRING_CONF,
-            DatagenConnector.SCHEMA_SOURCE_ERR
-        )
+        hasValidationError(DatagenConnectorConfig.SCHEMA_STRING_CONF, 1)
     );
     assertThat(
         validated,
-        hasValidationError(
-            DatagenConnectorConfig.SCHEMA_FILENAME_CONF,
-            DatagenConnector.SCHEMA_SOURCE_ERR
-        )
+        hasValidationError(DatagenConnectorConfig.SCHEMA_FILENAME_CONF, 1)
     );
   }
 
   @Test
   public void shouldFailValidationWithNoSchemaSources() {
     clearSchemaSources();
-
     Config validated = connector.validate(config);
 
     for (String k : DatagenConnectorConfig.schemaSourceKeys()) {
-      assertThat(validated, hasValidationError(k, DatagenConnector.SCHEMA_SOURCE_ERR));
+      assertThat(validated, hasValidationError(k, 1));
     }
   }
 
+  @Test(expected = ConfigException.class)
+  public void shouldFailValidationWithInvalidSchemaString() {
+    clearSchemaSources();
+    config.put(DatagenConnectorConfig.SCHEMA_STRING_CONF, "a schema");
+    new DatagenConnectorConfig(config);
+  }
+
+  @Test(expected = ConfigException.class)
+  public void shouldFailValidationWithInvalidFileName() {
+    clearSchemaSources();
+    config.put(DatagenConnectorConfig.SCHEMA_FILENAME_CONF, "a file name");
+    new DatagenConnectorConfig(config);
+  }
+
+  @Test(expected = ConfigException.class)
```

Comment:
Instead of configException , we can have the assertError mapping of connector config with the error string , somewhat similar to [this](https://github.com/confluentinc/kafka-connect-confluent-metrics/blob/master/kafka-connect-datadog-metrics/src/test/java/io/confluent/connect/datadog/metrics/DatadogMetricsSinkValidationTest.java#L220) one.